diff --git a/docroot/sites/all/modules/contrib/autosave/autosave.js b/docroot/sites/all/modules/contrib/autosave/autosave.js
index 77bc47b..6209b48 100644
--- a/docroot/sites/all/modules/contrib/autosave/autosave.js
+++ b/docroot/sites/all/modules/contrib/autosave/autosave.js
@@ -12,6 +12,7 @@ Drupal.behaviors.autosave.attach = function (context, settings) {
   }
 
   if (undefined !== settings.autosave) {
+    $("[name='autosave_form_path']").closest("form").css("opacity", 1);
     autosaveSettings = settings.autosave;
     var $selector = $('input[name="form_id"][value="' + autosaveSettings.form_id + '"] ').parents('form').not('.autosave-processed');
     $selector.addClass('autosave-processed').autosave({
@@ -40,6 +41,7 @@ Drupal.behaviors.autosave.attach = function (context, settings) {
               return false;
             });
             $('#autosave-status .restore-link').click(function(e) {
+              $("[name='autosave_form_path']").closest("form").css("opacity", 0.4);
               showingRestoreCommand = false;
               Drupal.behaviors.autosave.hideMessage();
             });
