diff --git a/docroot/sites/all/modules/contrib/autosave/autosave.module b/docroot/sites/all/modules/contrib/autosave/autosave.module
index 976246f..36881e0 100644
--- a/docroot/sites/all/modules/contrib/autosave/autosave.module
+++ b/docroot/sites/all/modules/contrib/autosave/autosave.module
@@ -305,6 +305,8 @@ function autosave_restore($form_id, $timestamp) {
     // Stop recording of form error messages
     $form_state['triggering_element']['#limit_validation_errors'] = array();
 
+    // Unset form_token from saved formto avoid outdated form between sessions.
+    unset($form_state['input']['form_token']);
     // When restoring we will need to know the form token so that the user can
     // be validated.
     $form = drupal_build_form($form_id, $form_state);
