diff --git a/modules/image/image.module b/modules/image/image.module
index ff50452..5cdd1b3 100644
--- a/modules/image/image.module
+++ b/modules/image/image.module
@@ -803,12 +803,7 @@ function image_style_deliver($style, $scheme) {
   if (!file_exists($derivative_uri)) {
     $lock_acquired = lock_acquire($lock_name);
     if (!$lock_acquired) {
-      // Tell client to retry again in 3 seconds. Currently no browsers are known
-      // to support Retry-After.
-      drupal_add_http_header('Status', '503 Service Unavailable');
-      drupal_add_http_header('Retry-After', 3);
-      print t('Image generation in progress. Try again shortly.');
-      drupal_exit();
+      lock_wait($lock_name);
     }
   }
 
