diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index f6ffea5..3c99165 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -23,7 +23,7 @@ define('DRUPAL_MINIMUM_PHP', '5.2.4');
 /**
  * Minimum recommended value of PHP memory_limit.
  */
-define('DRUPAL_MINIMUM_PHP_MEMORY_LIMIT', '32M');
+define('DRUPAL_MINIMUM_PHP_MEMORY_LIMIT', '256M');
 
 /**
  * Error reporting level: display no errors.
