--- a/.htaccess	2014-04-28 08:51:34.000000000 +0200
+++ b/.htaccess	2014-04-28 10:05:40.000000000 +0200
@@ -63,6 +63,19 @@
   RewriteCond %{HTTPS} on
   RewriteRule ^ - [E=protossl:s]
 
+  # Also set "protossl" for X-Forwarded-Proto HTTP header.
+  RewriteCond %{HTTP:X-Forwarded-Proto} =https
+  RewriteRule ^ - [E=protossl:s]
+
+  # Rewrite drd7stg.prod.acquia-sites.com,
+  # drd7XX.prod.acquia-sites.com, etc. hostnames to the *-acquia.dr.dk
+  # equivalent.
+  RewriteCond %{HTTP_HOST} ^drd7stg\.prod\.acquia-sites\.com
+  RewriteRule ^(.*)$ http%{ENV:protossl}://stage-acquia.dr.dk/$1 [R=301,L]
+
+  RewriteCond %{HTTP_HOST} ^drd7(.+)\.prod\.acquia-sites\.com
+  RewriteRule ^(.*)$ http%{ENV:protossl}://%1-acquia.dr.dk/$1 [R=301,L]
+
   # Make sure Authorization HTTP header is available to PHP
   # even when running as CGI or FastCGI.
   RewriteRule ^ - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
