diff --git a/modules/simpletest/drupal_web_test_case.php b/modules/simpletest/drupal_web_test_case.php
index 0853c7d..b00357d 100644
--- a/modules/simpletest/drupal_web_test_case.php
+++ b/modules/simpletest/drupal_web_test_case.php
@@ -1748,8 +1748,8 @@ class DrupalWebTestCase extends DrupalTestCase {
     // to reject the request citing "400 - Bad Request". To prevent
     // this, we strip the fragment from the request.
     // TODO: Remove this for Drupal 8, since fixed in curl 7.20.0.
+    $original_url = $curl_options[CURLOPT_URL];
     if (!empty($curl_options[CURLOPT_URL]) && strpos($curl_options[CURLOPT_URL], '#')) {
-      $original_url = $curl_options[CURLOPT_URL];
       $curl_options[CURLOPT_URL] = strtok($curl_options[CURLOPT_URL], '#');
     }
 
@@ -1763,6 +1763,12 @@ class DrupalWebTestCase extends DrupalTestCase {
       // not overwritten by Curl.
       $curl_options[CURLOPT_HTTPHEADER][] = 'Expect:';
     }
+
+    // Provide alter possibility for working around nginx probem.
+    if (function_exists('simpletest_curl_alter')) {
+      simpletest_curl_alter($curl_options);
+    }
+
     curl_setopt_array($this->curlHandle, $this->additionalCurlOptions + $curl_options);
 
     if (!$redirect) {
