diff --git a/docroot/sites/all/modules/contrib/hierarchical_select/hierarchical_select.module b/docroot/sites/all/modules/contrib/hierarchical_select/hierarchical_select.module
index ad59a9b..338f830 100644
--- a/docroot/sites/all/modules/contrib/hierarchical_select/hierarchical_select.module
+++ b/docroot/sites/all/modules/contrib/hierarchical_select/hierarchical_select.module
@@ -765,6 +765,14 @@ function form_hierarchical_select_process($element, &$form_state, $complete_form
   $hs_selection = $db_selection = array();
   list($hs_selection, $db_selection) = _hierarchical_select_process_calculate_selections($element, $hsid, $form_state);
 
+  drupal_add_js(array(
+    'hierarchical_select_values' => array(
+      'hsid' => $hsid,
+      'selected_all' => $db_selection,
+      'selected_last' => $hs_selection,
+    ),
+  ), 'setting');
+
   // Developer mode logging: log selections.
   _hs_process_developer_mode_log_selections($config, $hs_selection, $db_selection);
 
@@ -2147,8 +2155,6 @@ function _hierarchical_select_dropbox_generate($config, $selection) {
       }
     }
 
-    usort($dropbox->lineages, '_hierarchical_select_dropbox_sort');
-
     // Now store each lineage's selection too. This is needed on the client side
     // to enable the remove button to let the server know which selected items
     // should be removed.
