diff --git a/manualcrop.helpers.inc b/manualcrop.helpers.inc
index 3d34b23..9d8848d 100644
--- a/manualcrop.helpers.inc
+++ b/manualcrop.helpers.inc
@@ -151,7 +151,9 @@ function manualcrop_instance_required_styles($settings) {
 function manualcrop_croptool_process(&$form, &$form_state, &$element, $file, $settings = array(), $instance = NULL, $preview = 'preview') {
   static $processed_forms;
 
-  if (user_access('use manualcrop') && (!empty($_COOKIE['has_js']) || variable_get('manualcrop_skip_js_check', FALSE))) {
+  if (user_access('use manualcrop') &&
+    (strpos($element['#file']->filemime, 'image') !== FALSE) &&
+    (!empty($_COOKIE['has_js']) || variable_get('manualcrop_skip_js_check', FALSE))) {
     $styles = manualcrop_styles_with_crop();
 
     if (!empty($styles)) {
@@ -253,6 +255,12 @@ function manualcrop_croptool_process(&$form, &$form_state, &$element, $file, $se
         }
       }
 
+      // Add class to the wrapper.
+      if (empty($element['#attributes']['class'])) {
+        $element['#attributes']['class'] = array();
+      }
+      $element['#attributes']['class'][] = 'manualcrop-image-widget';
+
       // Add the dependencies and croptool.
       $js_identifier = _manualcrop_js_identifier(is_array($instance) ? $instance : $file);
       _manualcrop_attach_dependencies($element, $form_state, $js_identifier, $settings);
