diff --git a/oembedcore.module b/oembedcore.module
index 105e166..bf63f80 100644
--- a/oembedcore.module
+++ b/oembedcore.module
@@ -166,7 +166,7 @@ http://snd.sc/*';
   $provider->disabled = FALSE; /* Edit this to true to make a default provider disabled initially */
   $provider->name = 'twitter';
   $provider->title = 'Twitter';
-  $provider->endpoint = 'https://api.twitter.com/1/statuses/oembed.json';
+  $provider->endpoint = 'https://api.twitter.com/1.1/statuses/oembed.json';
   $provider->scheme = 'http://twitter.com/*/status/*
 http://twitter.com/*/statuses/*
 http://www.twitter.com/*/status/*
@@ -193,19 +193,21 @@ https://mobile.twitter.com/*/statuses/*';
   $provider->disabled = FALSE; /* Edit this to true to make a default provider disabled initially */
   $provider->name = 'vimeo';
   $provider->title = 'Vimeo';
-  $provider->endpoint = 'http://vimeo.com/api/oembed.json';
-  $provider->scheme = 'http://www.vimeo.com/groups/*/videos/*
-http://www.vimeo.com/*
+  $provider->endpoint = 'https://vimeo.com/api/oembed.json';
+  $provider->scheme = 'http://vimeo.com/*
+http://vimeo.com/channels/*/*
 http://vimeo.com/groups/*/videos/*
-http://vimeo.com/*
-http://vimeo.com/m/#/*';
+http://vimeo.com/m/*
+https://vimeo.com/*
+https://vimeo.com/channels/*/*
+https://vimeo.com/groups/*/videos/*';
   $providers['vimeo'] = $provider;
 
   $provider = new stdClass;
   $provider->disabled = FALSE; /* Edit this to true to make a default provider disabled initially */
   $provider->name = 'youtube';
   $provider->title = 'YouTube';
-  $provider->endpoint = 'http://www.youtube.com/oembed';
+  $provider->endpoint = 'https://www.youtube.com/oembed';
   $provider->scheme = 'http://*.youtube.com/watch*
 http://*.youtube.com/v/*
 https://*.youtube.com/watch*
