diff --git a/docroot/sites/all/modules/contrib/module_peytz_mail/modules/peytz_mail_subscribe/peytz_mail_subscribe.module b/docroot/sites/all/modules/contrib/module_peytz_mail/modules/peytz_mail_subscribe/peytz_mail_subscribe.module
index 1926cae..94fb9bf 100644
--- a/docroot/sites/all/modules/contrib/module_peytz_mail/modules/peytz_mail_subscribe/peytz_mail_subscribe.module
+++ b/docroot/sites/all/modules/contrib/module_peytz_mail/modules/peytz_mail_subscribe/peytz_mail_subscribe.module
@@ -50,7 +50,7 @@ function peytz_mail_subscribe_form($form, &$form_state, $conf) {
     return;
   }
 
-  $form['#attributes']['class'] = 'signup-form';
+  $form['#attributes']['class'][] = 'signup-form';
   $form['#attributes']['id'] = 'signup-form';
 
   if (!empty($conf['header'])) {
